sigc++/tuple-utils/tuple_end.h

```diff
@@ -56,6 +56,7 @@ tuple_end(T&& t) {
   static_assert(len <= size, "The tuple size must be less than or equal to the length.");
 
   if constexpr(len == 0) {
+    static_cast<void>(t);
```

Comment:
Please add a comment in the code about why this is here. One day we might be able to remove it, if we have that clue.